Add go2rtc (#34998)
home-assistant/home-assistant.github.io: source/_integrations/go2rtc.markdown

---
title: go2rtc
description: Instructions on how to setup go2rtc in Home Assistant.
ha_config_flow: true
ha_release: 2024.11
ha_iot_class: Local Polling
ha_codeowners:
  - '@home-assistant/core'
ha_domain: go2rtc
related:
  - docs: /installation/
---

The go2rtc {% term integration %} will connect to a go2rtc instance and will provide a WebRTC proxy for all your cameras.

> {% include integrations/config_flow.md %}

Reviewer: We've removed the config flow now.

If you are running Home Assistant with

- {% term "Home Assistant Operating System" %}
- {% term "Home Assistant Supervised" %}
- {% term "Home Assistant Container" %}
    

> than the go2rtc binary is already preinstalled in the container and the config flow will pick it automatically.

Reviewer: ```suggestion
the go2rtc binary is already preinstalled in the container and the config flow will pick it automatically.
```

> If you are running {% term "Home Assistant Core" %}, than you need to setup a go2rtc instance and provide the URL in the config flow.

Reviewer: ```suggestion
If you are running {% term "Home Assistant Core" %}, you need to setup a go2rtc instance and provide the URL in the config flow.
```

Reviewer: We should explain a bit more what URL this is.

---
title: go2rtc
description: Instructions on how to setup go2rtc in Home Assistant.
ha_category:
  - Camera
ha_release: 2024.11
ha_iot_class: Local Polling
ha_codeowners:
  - '@home-assistant/core'
ha_domain: go2rtc
ha_integration_type: system
related:
  - docs: /installation/
---

> The **go2rtc** {% term integration %} connects to a go2rtc instance and provides a WebRTC proxy for all your cameras. To learn more about go2rtc,  refer to the [project's GitHub page](https://github.com/AlexxIT/go2rtc/).

Reviewer 2: ```suggestion
go2rtc is an open source project providing a camera streaming application that supports formats such as RTSP, WebRTC, HomeKit, FFmpeg, RTMP.  The **go2rtc** {% term integration %} connects to a go2rtc instance and provides a WebRTC proxy for all your cameras. To learn more about go2rtc, refer to the [project's GitHub page](https://github.com/AlexxIT/go2rtc/).
```
Maybe we could add some context?

If you are using the [`default_config`](/integrations/default_config/) and run Home Assistant using one of the following installation methods, the go2rtc integration will be set up automatically and you don't need to do anything:

- {% term "Home Assistant Operating System" %}
- {% term "Home Assistant Supervised" %}
- {% term "Home Assistant Container" %}

## Configuration

This integration is part of the [`default_config`](/integrations/default_config/).

The following YAML options are available:

{% configuration %}
debug_ui:
  required: false
  description: Enables the UI of the go2rtc, which helps debugging WebRTC issues. The `debug_ui` should only be enabled during debugging as it will expose port 1984 without any authentication!
  default: false
  type: boolean
url:
  required: false
  description: The URL to the self-hosted [go2rtc](https://github.com/AlexxIT/go2rtc/) server
  type: string
{% endconfiguration %}

{% warning %}

> The `debug_ui` should only enabled during debugging as it will expose port 1984 without any authentication!

Reviewer: ```suggestion
The `debug_ui` should only be enabled during debugging as it will expose port 1984 without any authentication!
```

If you are using the [`default_config`](/integrations/default_config/) and run Home Assistant using one of the following installation methods, the go2rtc integration will be set up automatically and you don't need to do anything:

- {% term "Home Assistant Operating System" %}
- {% term "Home Assistant Supervised" %}
- {% term "Home Assistant Container" %}

## Configuration

This integration is part of the [`default_config`](/integrations/default_config/).

The following YAML options are available:

{% configuration %}
debug_ui:
  required: false
  description: Enables the UI of the go2rtc, which helps debugging WebRTC issues. The `debug_ui` should only be enabled during debugging as it will expose port 1984 without any authentication!
  default: false
  type: boolean
url:
  required: false
  description: The URL to the self-hosted [go2rtc](https://github.com/AlexxIT/go2rtc/) server
  type: string
{% endconfiguration %}

{% warning %}

The `debug_ui` should only enabled during debugging as it will expose port 1984 without any authentication!
Please disable the `debug_ui` immediately after debugging.

{% endwarning %}

> 
> 

Reviewer 2: ```suggestion
```

---
title: go2rtc
description: Instructions on how to setup go2rtc in Home Assistant.
ha_category:
  - Camera
ha_release: 2024.11
ha_iot_class: Local Polling
ha_codeowners:
  - '@home-assistant/core'
ha_domain: go2rtc
ha_integration_type: system
related:
  - docs: /installation/
---

> The go2rtc {% term integration %} will connect to a go2rtc instance and will provide a WebRTC proxy for all your cameras.

Reviewer 2: ```suggestion
The **go2rtc** {% term integration %} connects to a go2rtc instance and provides a WebRTC proxy for all your cameras. To learn more about go2rtc,  refer to the [project's GitHub page](https://github.com/AlexxIT/go2rtc/).

```

> If you are using the [`default_config`](/integrations/default_config/) and Home Assistant runs as

Reviewer 2: ```suggestion
If you are using the [`default_config`](/integrations/default_config/) and run Home Assistant using one of the following installation methods, the go2rtc integration will be set up automatically and you don't need to do anything:
```

- {% term "Home Assistant Operating System" %}
- {% term "Home Assistant Supervised" %}
- {% term "Home Assistant Container" %}

>     
> the go2rtc integration will be set up automatically and you don't need to do anything.
> 

Reviewer 2: ```suggestion
```

## Configuration

This integration is part of the [`default_config`](/integrations/default_config/).

The following YAML options are available:

{% configuration %}
debug_ui:
  required: false
  description: Enables the UI of the go2rtc, which helps debugging WebRTC issues. The `debug_ui` should only enabled during debugging as it will expose port 1984 without any authentication!
  default: false
  type: boolean
url:
  required: false
  description: The URL to the selfhosted [go2rtc](https://github.com/AlexxIT/go2rtc/) server
  type: string
{% endconfiguration %}

{% warning %}

The `debug_ui` should only enabled during debugging as it will expose port 1984 without any authentication!
Please disable the `debug_ui` immediately after debugging.

{% endwarning %}

> More information about the go2rtc project can be found on GitHub: https://github.com/AlexxIT/go2rtc/

Reviewer: ```suggestion
More information about the go2rtc project can be found on its [GitHub page](https://github.com/AlexxIT/go2rtc/)
```

> More information about the go2rtc project can be found on its [GitHub page](https://github.com/AlexxIT/go2rtc/)

Reviewer 2: ```suggestion
```

The go2rtc {% term integration %} will connect to a go2rtc instance and will provide a WebRTC proxy for all your cameras.

If you are using the [`default_config`](/integrations/default_config/) and Home Assistant runs as

- {% term "Home Assistant Operating System" %}
- {% term "Home Assistant Supervised" %}
- {% term "Home Assistant Container" %}
    
the go2rtc integration will be set up automatically and you don't need to do anything.

## Configuration

This integration is part of the [`default_config`](/integrations/default_config/).

The following YAML options are available:

{% configuration %}
debug_ui:
  required: false

>   description: Enables the UI of the go2rtc, which helps debugging WebRTC issues. The `debug_ui` should only enabled during debugging as it will expose port 1984 without any authentication!

Reviewer: ```suggestion
  description: Enables the UI of the go2rtc, which helps debugging WebRTC issues. The `debug_ui` should only be enabled during debugging as it will expose port 1984 without any authentication!
```

  default: false
  type: boolean
url:
  required: false

>   description: The URL to the selfhosted [go2rtc](https://github.com/AlexxIT/go2rtc/) server

Reviewer: ```suggestion
  description: The URL to the self-hosted [go2rtc](https://github.com/AlexxIT/go2rtc/) server
```

  type: string
{% endconfiguration %}

### Examples

> Use a selfhosted instance:

Reviewer: ```suggestion
Use a self-hosted instance:
```
